drivers/src/dma.rs

```diff
@@ -563,6 +572,11 @@ impl<'a> DmaRecovery<'a> {
         mmio.check_error(t)
     }
 
+    fn is_i3c(&self, addr: AxiAddr) -> bool {
+        let i3c = self.recovery_base + Self::INDIRECT_FIFO_DATA_OFFSET;
+        i3c.hi == addr.hi && i3c.lo == addr.lo
```

Comment:
Surprised the `AxiAddr` type doesn't support `PartialEq` & `Eq`

Reply:
Yeah, not sure why I didn't add those. Let me go ahead and fix that and simplify this.